cedar-policy-core/src/entities.rs

```diff
@@ -96,12 +96,12 @@ impl Entities {
             None => match self.mode {
                 Mode::Concrete => Dereference::NoSuchEntity,
                 #[cfg(feature = "partial-eval")]
-                Mode::Partial => Dereference::Residual(Expr::unknown_with_type(
-                    format!("{uid}"),
-                    Some(Type::Entity {
+                Mode::Partial => Dereference::Residual(Expr::unknown(Unknown {
+                    name: format!("{uid}").into(),
+                    type_annotation: Some(Type::Entity {
                         ty: uid.entity_type().clone(),
                     }),
-                )),
+                })),
```

Comment:
Why make this manually instead of using a constructor?

Reply:
No particular reason (although the named arguments you get this way are nice).  I'll change

Reply:
Changed